torch/utils/hipify/cuda_to_hip_mappings.py

```diff
@@ -64,17 +64,17 @@
 # List of math functions that should be replaced inside device code only.
 MATH_TRANSPILATIONS = collections.OrderedDict(
     [
-        ("std::max", ("::max")),
-        ("std::min", ("::min")),
-        ("std::ceil", ("::ceil")),
-        ("std::floor", ("::floor")),
-        ("std::exp", ("::exp")),
-        ("std::log", ("::log")),
-        ("std::pow", ("::pow")),
-        ("std::fabs", ("::fabs")),
-        ("std::fmod", ("::fmod")),
-        ("std::remainder", ("::remainder")),
-        ("std::frexp", ("::frexp")),
+        ("std::max", ("std::max")),
+        ("std::min", ("std::min")),
+        ("std::ceil", ("std::ceil")),
+        ("std::floor", ("std::floor")),
+        ("std::exp", ("std::exp")),
+        ("std::log", ("std::log")),
+        ("std::pow", ("std::pow")),
+        ("std::fabs", ("std::fabs")),
+        ("std::fmod", ("std::fmod")),
+        ("std::remainder", ("std::remainder")),
+        ("std::frexp", ("std::frexp")),
```

Comment:
For these math operations we do not want to use the std implementations of them but instead the implementations provided by HIP.  That's what I recall.

Reply:
Are we facing problems by not mapping them to std:: namespace?

Reply:
@jeffdaily Have tested on the rocm6.0 with nightly pytorch build and rocm6.2 provided by @jithunnair-amd, the hip can't handle it properly.  got some error like this "/root/AISS-Data-Science/mamba/causal-conv1d/csrc/hip/causal_conv1d_fwd.hip:36:11: error: no matching function for call to 'max'
        : ::max({sizeof(typename BlockLoadT::TempStorage), sizeof(typename BlockStoreT::TempStorage)});"

Reply:
It can be reproduced by this ticket https://ontrack-internal.amd.com/browse/SWDEV-458189

Reply:
I was working on a similar issue recently. It appears that these std mappings are handled differently depending on whether we are talking about device vs host functions. I am not sure how it will behave in cases when a function is defined as both, as it is in https://github.com/state-spaces/mamba/blob/main/csrc/selective_scan/uninitialized_copy.cuh, for example.

Maybe the parser gets confused for such functions?

Reply:
I suspect it might be due to a function being defined as both __host__ and __device__, while these mappings are supposed to be replaced in device function only. Maybe the hipifyer does not handle the "both" case properly?